src/RobloxRenderer.spec.lua

```diff
@@ -686,6 +687,100 @@ return function()
 		end)
 	end)
 
+	describe("Fragments", function()
+		it("should parent the fragment's elements into the fragment's parent", function()
+			local hostParent = Instance.new("Folder")
+
+			local fragment = createFragment({
+				key = createElement("IntValue", {
+					Value = 1,
+				})
+			})
+
+			local node = reconciler.mountVirtualNode(fragment, hostParent, "test")
+
+			expect(hostParent:FindFirstChild("key")).to.be.ok()
+			expect(hostParent.key.ClassName).to.equal("IntValue")
+			expect(hostParent.key.Value).to.equal(1)
+
+			reconciler.unmountVirtualNode(node)
+
+			expect(#hostParent:GetChildren()).to.equal(0)
+		end)
+
+		it("should allow sibling fragment to have common keys", function()
+			local hostParent = Instance.new("Folder")
+			local hostKey = "Test"
+
+			local childAComponentSpy = createSpy(function(props)
+				return createElement("StringValue", {
+					Value = "A",
+				})
+			end)
+			local childBComponentSpy = createSpy(function(props)
+				return createElement("StringValue", {
+					Value = "B",
+				})
+			end)
```

Comment:
Do these needs to be spies?

Reply:
I don't think they need, I could have asserted that they were called, but instead I removed them and asserted that the values are correctly created